back/spec/services/url_validation_service_spec.rb

```diff
@@ -0,0 +1,89 @@
+require 'rails_helper'
+
+describe UrlValidationService do
+  subject(:service) { described_class.new }
+
+  describe 'whitelist verification' do
```

Comment:
I failed to see it during PP:

```suggestion
  describe '#url_whitelisted?' do
```

Reply:
Ah true - I will add in those tests.

Reply:
I did not mention it, but writing `describe '#url_whitelisted?' do` is a pattern to say "here are the tests for the method `url_whitelisted?` in the described class".

Reply:
Ok that makes sense - thanks!